config/import-control.xml

```diff
@@ -7,7 +7,18 @@
     <allow pkg="java.io"/>
     <allow pkg="java.util"/>
     <allow pkg="com.github.checkstyle.regression.data"/>
+    <allow pkg="com.github.checkstyle.regression.util"/>
+    <allow pkg="org.immutables"/>
```

Comment:
I believe there would be many fields in `ModuleInfo` in the future. 
If we use constructor to pass these parameters, it would be ugly. 《Effective Java》 said it is better to use builder style if constructor has more than three parameters.
Another annoying thing is the `equals` method. We would have a lot of `equals` asserting in UTs. It is hard to appease codecov if we write this manually. Supposed there are 5 fields then the `equals` method would have 32 branches(2^5). That's horrible. If we write a `equals` method in UTs, yeah that could be a workaround, be a little bit strange. And it is also bad that when we add or change some fields, then we have to update the utility `equals` in UTs.

So I think it is not bad if an annotation processing tool could generate these for us. We could then focus on more important things but not writing builders, equals and thinking how to appease codecov.